include/bout/fv_ops.hxx

```diff
@@ -344,10 +343,10 @@ namespace FV {
    */
   template<typename CellEdges = MC>
   const Field3D Div_f_v(const Field3D &n_in, const Vector3D &v, bool bndry_flux) {
-    ASSERT2(n_in.getLocation() == v.getLocation());
+    ASSERT1(n_in.getLocation() == v.getLocation());
+    ASSERT1(areFieldsCompatible(n_in, v.x));
```

Comment:
Do we also need the location check on `v` if we're checking `v.x`?

Reply:
It does: the location check on `v` makes sure its location isn't `CELL_VSHIFT`, which isn't supported by this function.